geometric_sqrt.go

```diff
@@ -0,0 +1,51 @@
+package interpolator
+
+import (
+	"fmt"
+	"math"
+)
+
+// GeometricSqrt is a geometric interpolator scaled with a square root.
+type GeometricSqrt struct {
+	xys XYs
+}
+
+// NewGeometricSqrt builds a geometric sqrt interpolator.
+// The input `xys` must be ordered, have unique abscissas
+// and positive ordinates.
+func NewGeometricSqrt(xys XYs) (*GeometricSqrt, error) {
+	if l := len(xys); l < 2 {
+		return nil, fmt.Errorf("at least 2 points are required to build a geometric sqrt interpolator, but got %d", l)
+	}
+	for _, xy := range xys {
+		if xy.Y < epsilon {
+			return nil, fmt.Errorf("input xys must have non-negative ordinates")
+		}
+	}
+	return &GeometricSqrt{
+		xys: xys,
+	}, nil
+}
+
+// Value compute the value of f(x) based on geometric sqrt interpolation with threshold.
+func (interp GeometricSqrt) Value(x float64) float64 {
+	p1, p2 := interp.xys.Interval(x)
+	if x <= p1.X {
+		return p1.Y
+	}
+	if x >= p2.X {
+		return p2.Y
+	}
+	lambda := math.Sqrt((x - p1.X) / (p2.X - p1.X))
+	return math.Pow(p1.Y, (1.0-lambda)) * math.Pow(p2.Y, lambda)
```

Comment:
Although less safe, as it implies a division, you may rather apply `Pow` just once:
```go
return p1.Y * math.Pow(p2.Y/p1.Y, lambda)
```
Maybe apply the same logic as in the classic geometric interpolator (but for other reasons it's better as you did, since you don't have to perform further input checks).

Reply:
> apply the same logic as in the classic geometric interpolator

That's what I did, no ?